### PR TITLE
[DistillationTrainer refactor] Convention/cosmetic sweep

### PR DESCRIPTION
**Pull request by qgallouedec**
*Item 46 (Group F) — stacked on 44. Part of #6449. Cosmetic-only, no behaviour change.*

The "down to cosmetic details" pass, aligning the trainer with GRPO's conventions:
- **Class docstring** rewritten to GRPO's `Example:` + `Args:` format (was a bullet list). Shared-param docstrings (`model`, `args`, `train_dataset`, `eval_dataset`, `processing_class`, `callbacks`, `optimizers`, `quantization_config`, `peft_config`) copied verbatim from `grpo_trainer.py`; `teacher_model` written for distillation; runnable `Example:` block.
- **Box dividers removed** — the 4 `# ────…` method-section header blocks and the inline `# ── X ──` decorations (GRPO uses plain `# X` comments, no `──`).
- **`@profiling_decorator` on `compute_loss`** (GRPO has it there; distillation was missing it — now `_get_last_hidden_state`/`_prepare_inputs`/`compute_loss`, matching GRPO).
- **Imports consolidated** to GRPO's single `from transformers import (...)` (was 7 deep-submodule imports); `processing_class` type hint narrowed to GRPO's `PreTrainedTokenizerBase | ProcessorMixin | None`, dropping the now-unused `BaseImageProcessor`/`FeatureExtractionMixin`.
- **Name-drops:** trimmed gratuitous "GRPO" mentions (`(as GRPO does)`, "Mirrors GRPO's `compute_liger_loss`", a stale "Unwired until the GRPO generation stack lands"); **kept** the load-bearing ones that explain a deliberate divergence or slot-in (the `generation_batch_size` slot-in, the `_generate_and_score_completions` naming rationale).

Verified: ruff clean; full distillation + server suites 54 green.

---

> [!NOTE]
> **Low Risk**
> Mostly docs, comments, imports, and internal renaming; profiling decorators only affect observability. Default output directory when `args=None` changes naming but not loss or training flow.
> 
> **Overview**
> Aligns **`DistillationTrainer`** (and **`ServerDistillationTrainer`**) with **`GRPOTrainer`** style without changing training logic.
> 
> **Documentation & API surface:** Replaces the bullet-list class docstring with GRPO’s **`Example:`** / **`Args:`** layout (shared parameter docs plus distillation-specific **`teacher_model`**). Tightens **`__init__`** type hints (`PeftModel`, narrower **`processing_class`**) and uses **`get_config_model_id`** for model paths; when **`args`** is omitted, default **`DistillationConfig`** output dir becomes **`{model_basename}-Distillation`** instead of **`tmp_distillation`**.
> 
> **Code style:** Consolidates Transformers imports into one block, drops unused **`torch.nn`** / image-processor imports, swaps decorative **`# ── … ──`** section headers for plain comments, and trims redundant GRPO name-drops in comments.
> 
> **GRPO parity:** Renames the internal Liger flag **`use_liger_loss` → `use_liger_kernel`** (tests updated). Adds **`@profiling_decorator`** on **`DistillationTrainer.compute_loss`** and **`ServerDistillationTrainer.compute_loss`** so profiling matches GRPO’s hot paths.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 420e4fa868fb3b726535baccc9aa13cdcbbb3516. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/distillation/distillation_trainer.py [hunk 1]
```diff
@@ -281,14 +286,79 @@ def _chunked_divergence_loss(
...
+    >>> trainer = DistillationTrainer(
+    ...     model="Qwen/Qwen2.5-0.5B-Instruct",
+    ...     teacher_model="Qwen/Qwen2.5-1.5B-Instruct",
+    ...     train_dataset=dataset,
+    ... )
+    >>> trainer.train()
+    ```
+
+    Args:
+        model (`str` or [`~transformers.PreTrainedModel`] or [`~peft.PeftModel`]):
```
Comment (cmpatino) [inline trl/experimental/distillation/distillation_trainer.py:310]: Nit: There's a mismatch between the docstring and the signature. The signature still states `[PreTrainedModel | nn.Module | str](model: PreTrainedModel | nn.Module | str)`

**Review by cmpatino — APPROVED** [review]
Non-blocking comment about a mismatch between the docstring and the signature.

File: trl/experimental/distillation/distillation_trainer.py (hunk 1 above)
Comment (qgallouedec) [reply trl/experimental/distillation/distillation_trainer.py:310]: correct, addressed in https://github.com/huggingface/trl/pull/6630/commits/420e4fa868fb3b726535baccc9aa13cdcbbb3516
